x-pack/plugins/stack_connectors/public/connector_types/inference/types.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { UserConfiguredActionConnector } from '@kbn/triggers-actions-ui-plugin/public/types';
+import { ActionTypeModel as ConnectorTypeModel } from '@kbn/triggers-actions-ui-plugin/public';
+import { SUB_ACTION } from '../../../common/inference/constants';
+import {
+  ChatCompleteParams,
+  RerankParams,
+  SparseEmbeddingParams,
+  TextEmbeddingParams,
+} from '../../../common/inference/types';
+import { ConfigProperties } from '../lib/dynamic_config/types';
+
+export type InferenceActionParams =
+  | { subAction: SUB_ACTION.COMPLETION; subActionParams: ChatCompleteParams }
+  | { subAction: SUB_ACTION.RERANK; subActionParams: RerankParams }
+  | { subAction: SUB_ACTION.SPARSE_EMBEDDING; subActionParams: SparseEmbeddingParams }
+  | { subAction: SUB_ACTION.TEXT_EMBEDDING; subActionParams: TextEmbeddingParams };
+
+export type FieldsConfiguration = Record<string, ConfigProperties>;
+
+export interface Config {
```

Comment:
The pattern I've seen folks do to access types from schema in `public`, is to keep the schema and it's types in `server`, and then `import type` those in the `public` code, right from `server`.  You can only do `import type` like that, `import` by itself won't work (well, it will cause problems, but I think it does kinda work).

That will ensure there's a single definition for these types.